specifyweb/frontend/js_src/lib/components/WorkBench/hooks.ts

```diff
@@ -124,21 +137,21 @@ export function getHotHooks(wbView: WbView) {
         ),
       ]);
       if (JSON.stringify(issues) !== JSON.stringify(newIssues))
-        wbView.cells.updateCellMeta(
+        workbench.cells!.updateCellMeta(
           physicalRow,
           physicalCol,
           'issues',
           newIssues
         );
     },
 
-    afterUndo: (data) => afterUndoRedo(wbView, 'undo', data),
+    afterUndo: (data) => afterUndoRedo(workbench, 'undo', data),
 
-    afterRedo: (data) => afterUndoRedo(wbView, 'redo', data),
+    afterRedo: (data) => afterUndoRedo(workbench, 'redo', data),
 
     beforePaste: () =>
-      !wbView.uploadedView &&
-      !wbView.isUploaded &&
+      // !wbView.uploadedView &&
+      // !wbView.isUploaded &&
```

Comment:
this was here to ensure that when dataset is readonly (i.e because it has already been uploaded) you can't do `Ctrl+V` anymore to paste some data - probably still needed

also regarding `wbView.uploadedView`, while upload results side-panel is open, the dataset should be read-only too